File: universidad/semestre(GTI)-1/practicas/practica-01/Función1/porTres.cpp
//--------------------------------------------------------------------------------------------
// Fichero: PorTres.cpp
// Fecha: 25/9/25
// Descripción: Programa que multiplica un número por tres a través de una función en C++.
// Autor: Carlos Galán Cubells
//--------------------------------------------------------------------------------------------
#include <iostream>
using namespace std;
//--------------------------------------------------------------------------------------------
// R -> porTres -> R
//--------------------------------------------------------------------------------------------
double porTres(double n) {
    return n * 3.0;
}
//--------------------------------------------------------------------------------------------

int main() {
    cout << "Hello world! \n";

    //Voy a llamar a la función
    //HACIENDO UNA PRUEBA AUTOMÁTICA
    double r;

    r = porTres(4.5);
    if (r != 13.5) {
        cout << "La función va mal \n";
    }

    //otra prueba automática
    r = porTres(7.0);
    if (r != 21.0) {
        cout << "La función va mal \n";
    }

    cout << "Hasta luego Lucas \n";

    return 0;
}
//--------------------------------------------------------------------------------------------

File: universidad/semestre(GTI)-2/practicas/practica-01/porTres.js
/*--------------------------------------------------------------------------------------------
Fichero: practica1.js
Autor: Carlos Galán Cubells
Fecha: 05/02/2026
Descripción: Función que recibe un número real y lo multiplica por tres.
--------------------------------------------------------------------------------------------*/
//------------------------------------------------------------------------------------------
//  Función por tres:   R -> porTres -> R
function porTres(a){

    //Multiplicara el número a por tres y nos lo devolverá:
    return a * 3;
}
//----------------------------------------------------------
//  Prueba: 

//  Le damos un valor a la variable a:
let a = 1;

//  Generamos una variable res que será la prueba para comprobar si funciona porTres:
let res = porTres(a);

//  Hacemos la comprobación y le decimos que si falla nos lo notifique por pantalla:
if(res != 3) console.log("Algo va mal");
//------------------------------------------------------------------------------------------